pkg/predicate/predicate.go

```diff
@@ -116,3 +116,32 @@ func (ResourceVersionChangedPredicate) Update(e event.UpdateEvent) bool {
 	}
 	return true
 }
+
+// GenerationChangedPredicate implements a default update predicate function on generation change
```

Comment:
It will be good to add the details that you specified in the PR description especially mentioning `status` sub-resource. That sort of explains `when` this predicate is useful.

Reply:
I would also like to see an update to the examples to show how to use the predicate? WDYT?

Reply:
Did you mean showing how to use one of the pre-defined predicates as an argument to the controller's Watch()?
I guess I could add that since I don't see it in any of the other examples:
https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/controller#ex-Controller

There's also an existing example in the predicate godoc, outlining how to define a new predicate.
Interestingly enough it's using generation change as the example.
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/predicate/example_test.go#L27-L33
https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/predicate#ex-Funcs

I could update that to be more complete like the one written here. Or leave that as it is if it's meant to be just a short example.

@droot Let me know what you think.

Reply:
I'd prefer to keep that example as-is.

In addition to a "why", I think this needs a "caveats" section -- like "this means you won't self-correct when someone else wipes your status".